### PR TITLE
Add manifest and requirements to fix PipI shipping

### PR DESCRIPTION
**Pull request by gcarleo**
This should (at least partially) fix most of the issues we have seen so far in #170. 
I have registered a new version 
```
pip3 install netket
```
should work, provided that CMake is available and some MPI compiler is available. For some applications numpy is also needed, but users who try to run those applications and don't have numpy will see an error message which is clear enough.

### CODE REVIEW

File: MANIFEST.in [hunk 1]
```diff
@@ -0,0 +1,5 @@
+include CMakeLists.txt
+recursive-include CMakeModules *
+include LICENSE
```
Comment (twesterhout) [inline MANIFEST.in]: Aren't you forgetting `README.md`?

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+cmake>3.1
+numpy>1.16
```
Comment (twesterhout) [inline requirements.txt]: What's wrong with versions 1.14 & 1.15 of numpy?

File: MANIFEST.in (hunk 1 above)
Comment (gcarleo) [reply MANIFEST.in]: It's already included by default, but I am going to add it now anyway

File: requirements.txt (hunk 2 above)
Comment (gcarleo) [reply requirements.txt]: good point, those work too

**Pull request merged**